src/gui/appclient.cpp

```diff
@@ -501,6 +489,21 @@ void AppClient::onQuit() {
     quit();
 }
 
+void AppClient::onServerDisconnected() {
+#if NDEBUG
+    if(connectToServer()){
+    		qCInfo(lcAppClient) << "Reconnected to server";
+    }else {
+			qCCritical(lcAppClient) << "Failed to reconnect to server";
+            //TODO: Implement a limit of reconnection trials per x minutes
+			startServerAndKillClient();
```

Comment:
What about `startServerAndDie`?

Reply:
Thanks you for the suggestion, updated in last commit.